core/src/main/java/org/elasticsearch/http/netty/NettyHttpRequest.java

```diff
@@ -69,6 +69,14 @@ public NettyHttpRequest(org.jboss.netty.handler.codec.http.HttpRequest request,
     @Override
     public Method method() {
         HttpMethod httpMethod = request.getMethod();
+
+        if (request.headers() != null) {
+            String methodOverride = request.headers().get("X-HTTP-Method-Override");
+            if (methodOverride != null) {
+                httpMethod = HttpMethod.valueOf(methodOverride);
+            }
+        }
+
```

Comment:
I'm ok with the updated approach limiting it to `POST` operations. The other thing I would like to see is making this configurable (as in be able to turn it off) and possibly part of the Netty pipeline after the `NettyHttpRequestDecoder`? Maybe off by default is the right choice here and we have to enable it for our tests?